pyccel/ast/core.py

```diff
@@ -2610,6 +2632,16 @@ def is_static(self):
         """
         return self._is_static
 
+    @property
+    def is_semantic(self):
+        """
+        Indicates if the function is annotated.
+
+        Indicates if the function has been annotated with type descriptors
+        in the semantic stage.
+        """
+        return self._is_semantic
```

Comment:
Should we not call this property `is_annotated`?

Reply:
it was before, but @EmilyBourne suggested using `is_semantic` so that we don't confuse it with type annotation if I remember correctly

Reply:
Yes. In Python "annotated" means that the function has type annotations. This is not what is meant here so I think it is important to avoid using potentially confusing terminology.

That said maybe the docstring could also be modified to clarify this?
```suggestion
    @property
    def is_semantic(self):
        """
        Indicates if the function was created with semantic information.

        Indicates if the function has been annotated with type descriptors
        in the semantic stage.
        """
        return self._is_semantic
```

Reply:
I agree, the docstring needs improving.